jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildAndCacheApplicationLayerStep.java

```diff
@@ -46,24 +50,54 @@ class BuildAndCacheApplicationLayerStep implements Callable<PreparedLayer> {
    */
   static ImmutableList<BuildAndCacheApplicationLayerStep> makeList(
       BuildContext buildContext, ProgressEventDispatcher.Factory progressEventDispatcherFactory) {
-    List<FileEntriesLayer> layerConfigurations = buildContext.getLayerConfigurations();
+    List<? extends LayerObject> layerConfigurations = buildContext.getLayerConfigurations();
+    int fileLayerCount = 0;
+    for (LayerObject layerConfiguration : layerConfigurations) {
+      if (layerConfiguration instanceof FileEntriesLayer) {
+        fileLayerCount++;
+      } else if (layerConfiguration instanceof PlatformDependentLayer) {
+        fileLayerCount += ((PlatformDependentLayer) layerConfiguration).getEntries().size();
+      }
+    }
 
     try (ProgressEventDispatcher progressEventDispatcher =
             progressEventDispatcherFactory.create(
-                "launching application layer builders", layerConfigurations.size());
+                "launching application layer builders", fileLayerCount);
         TimerEventDispatcher ignored =
             new TimerEventDispatcher(
                 buildContext.getEventHandlers(), "Preparing application layer builders")) {
       return layerConfigurations.stream()
           // Skips the layer if empty.
-          .filter(layerConfiguration -> !layerConfiguration.getEntries().isEmpty())
-          .map(
-              layerConfiguration ->
-                  new BuildAndCacheApplicationLayerStep(
-                      buildContext,
-                      progressEventDispatcher.newChildProducer(),
-                      layerConfiguration.getName(),
-                      layerConfiguration))
+          .filter(layerConfiguration -> {
+            if (layerConfiguration instanceof FileEntriesLayer) {
+              return !((FileEntriesLayer) layerConfiguration).getEntries().isEmpty();
+            } else if (layerConfiguration instanceof PlatformDependentLayer) {
+              return !((PlatformDependentLayer) layerConfiguration).getEntries().isEmpty();
+            }
+            return true;
+          })
+          .flatMap(layerConfiguration -> {
+            if (layerConfiguration instanceof FileEntriesLayer) {
+              return Stream.of(new BuildAndCacheApplicationLayerStep(
+                  buildContext,
+                  progressEventDispatcher.newChildProducer(),
+                  layerConfiguration.getName(),
+                  (FileEntriesLayer) layerConfiguration,
+                  new Platform("amd64", "linux")));
+            } else if (layerConfiguration instanceof PlatformDependentLayer) {
```

Comment:
Something key to understand with this is that a layers gets built for every platform in the  `PlatformDependentLayer`, and only later _might_ that layer be bound to an image. It's possible for layers to be prepared and to never get used. That would happen if the build spec has more platforms than Jib has been asked to produce images for.